frontend/src/scenes/feature-flags/JSONEditorInput.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react'
+import MonacoEditor from '@monaco-editor/react'
+import './JSONEditorInput.scss'
+import { JsonType } from '~/types'
+
+interface EditorProps {
+    onChange?: (val: string | undefined) => void
+    lineHeight?: number
+    defaultNumberOfLines?: number
+    value?: JsonType
+    readOnly?: boolean
+}
+
+export function JSONEditorInput({
+    onChange,
+    lineHeight = 20,
+    defaultNumberOfLines = 1,
+    value = '',
+    readOnly = false,
+}: EditorProps): JSX.Element {
+    const valString = value?.toString() || ''
+    const _lineHeight = lineHeight
+    const defaultLines = Math.max(defaultNumberOfLines, valString.split(/\r\n|\r|\n/).length) + 1
+    const defaultHeight = _lineHeight * defaultLines
+    const [height, setHeight] = useState(defaultHeight)
+
+    const updateHeight = (val: string | undefined): void => {
+        if (val) {
+            const lineCount = val.split(/\r\n|\r|\n/).length
+            const newLineCount = Math.max(lineCount, defaultNumberOfLines) + 1
+            setHeight(_lineHeight * newLineCount)
+        } else {
+            setHeight(_lineHeight * (defaultNumberOfLines + 1))
+        }
+    }
+
+    return (
+        <div className="hog-editor">
```

Comment:
Heads up we do have scss naming conventions, check out https://posthog.com/handbook/engineering/conventions/frontend-coding under `CSS`